### ISSUE TITLE
HackStudio: Find in files crashes with m_ptr[offset] >> 6 == 2 assertion failure

### ISSUE THREAD

**Issue by Lubrsi** [SerenityOS/serenity#5572]
Doesn't matter what you type, even blank causes it.

Trace:
```
HackStudio(33): USERSPACE(33) ASSERTION FAILED: m_ptr[offset] >> 6 == 2
../AK/Utf8View.cpp:206
CrashDaemon(20): --- Backtrace for thread #0 (TID 33) ---
CrashDaemon(20): 0x87ae6188: [libsystem.so]  +0x8 (syscall.cpp:35)
CrashDaemon(20): 0x9b2f5cb6: [libc.so] __assertion_failed +0x106 (syscall.h:44)
CrashDaemon(20): 0x9b3473cf: [libc.so] AK::Utf8CodepointIterator::operator*() const +0x24f (Utf8View.cpp:206)
CrashDaemon(20): 0x2bf7c58d: [libgui.so] GUI::TextDocumentLine::set_text(GUI::TextDocument&, AK::StringView const&) Shell(25): Job entry 'HackStudio' deleted in 8701 ms
+0x15d (Vector.h:176)
CrashDaemon(20): 0x2bf7c73e: [libgui.so] GUI::TextDocument::set_text(AK::StringView const&) +0xee (TextDocument.cpp:72)
CrashDaemon(20): 0x168cc4dd: [/bin/HackStudio] HackStudio::ProjectFile::document() const +0x19d (Atomic.h:255)
CrashDaemon(20): 0x168a2882: [/bin/HackStudio] AK::Function<void (HackStudio::ProjectFile const&)>::CallableWrapper<HackStudio::find_in_files(AK::StringView const&)::{lambda(auto:1&)#1}>::call(HackStudio::ProjectFile const&) const +0x42 (FindInFilesWidget.cpp:117)
CrashDaemon(20): 0x168cbdeb: [/bin/HackStudio] void HackStudio::traverse_model<HackStudio::Project::for_each_text_file(AK::Function<void (HackStudio::ProjectFile const&)>) const::{lambda(auto:1&)#1}>(GUI::FileSystemModel const&, HackStudio::Project::for_each_text_file(AK::Function<void (HackStudio::ProjectFile const&)>) const::{lambda(auto:1&)#1}::ModelIndex const&, HackStudio::Project::for_each_text_file(AK::Function<void (HackStudio::ProjectFile const&)>) const::{lambda(auto:1&)#1}) +0xdb (Function.h:57)
CrashDaemon(20): 0x168cbe76: [/bin/HackStudio] void HackStudio::traverse_model<HackStudio::Project::for_each_text_file(AK::Function<void (HackStudio::ProjectFile const&)>) const::{lambda(auto:1&)#1}>(GUI::FileSystemModel const&, HackStudio::Project::for_each_text_file(AK::Function<void (HackStudio::ProjectFile const&)>) const::{lambda(auto:1&)#1}::ModelIndex const&, HackStudio::Project::for_each_text_file(AK::Function<void (HackStudio::ProjectFile const&)>) const::{lambda(auto:1&)#1}) +0x166 (Project.cpp:58)
CrashDaemon(20): 0x168cbe76: [/bin/HackStudio] void HackStudio::traverse_model<HackStudio::Project::for_each_text_file(AK::Function<void (HackStudio::ProjectFile const&)>) const::{lambda(auto:1&)#1}>(GUI::FileSystemModel const&, HackStudio::Project::for_each_text_file(AK::Function<void (HackStudio::ProjectFile const&)>) const::{lambda(auto:1&)#1}::ModelIndex const&, HackStudio::Project::for_each_text_file(AK::Function<void (HackStudio::ProjectFile const&)>) const::{lambda(auto:1&)#1}) +0x166 (Project.cpp:58)
CrashDaemon(20): 0x168cbff8: [/bin/HackStudio] HackStudio::Project::for_each_text_file(AK::Function<void (HackStudio::ProjectFile const&)>) const +0x58 (Project.cpp:71)
CrashDaemon(20): 0x168a325f: [/bin/HackStudio] AK::Function<void (unsigned int)>::CallableWrapper<HackStudio::FindInFilesWidget::FindInFilesWidget()::{lambda(auto:1)#2}>::call(unsigned int) const +0x9f (OwnPtr.h:135)
CrashDaemon(20): 0x2beee794: [libgui.so] GUI::Button::click(unsigned int) +0xd4 (Function.h:57)
CrashDaemon(20): 0x168a27d1: [/bin/HackStudio] AK::Function<void ()>::CallableWrapper<HackStudio::FindInFilesWidget::FindInFilesWidget()::{lambda()#3}>::call() const +0x31 (Function.h:111)
CrashDaemon(20): 0x2bf8ad89: [libgui.so] GUI::TextEditor::keydown_event(GUI::KeyEvent&) +0x279 (Function.h:57)
CrashDaemon(20): 0x2bf7af01: [libgui.so] GUI::TextBox::keydown_event(GUI::KeyEvent&) +0x31 (Event.h:289)
CrashDaemon(20): 0x2bf9fbd1: [libgui.so] GUI::Widget::handle_keydown_event(GUI::KeyEvent&) +0x31 (Widget.cpp:300)
CrashDaemon(20): 0x2bfa50fa: [libgui.so] .L2863 +0xa (Widget.cpp:263)
CrashDaemon(20): 0x341e9ab4: [libcore.so] Core::Object::dispatch_event(Core::Event&, Core::Object*) +0x64 (Object.h:104)
CrashDaemon(20): 0x2bfad6ea: [libgui.so] GUI::Window::handle_key_event(GUI::KeyEvent&) +0xda (Window.cpp:433)
CrashDaemon(20): 0x2bfb45aa: [libgui.so] GUI::Window::event(Core::Event&) +0xea (Window.cpp:538)
CrashDaemon(20): 0x341e9ab4: [libcore.so] Core::Object::dispatch_event(Core::Event&, Core::Object*) +0x64 (Object.h:104)
CrashDaemon(20): 0x341d2254: [libcore.so] Core::EventLoop::pump(Core::EventLoop::WaitMode) +0x3e4 (Atomic.h:255)
CrashDaemon(20): 0x341d281b: [libcore.so] Core::EventLoop::exec() +0x4b (EventLoop.cpp:390)
CrashDaemon(20): 0x2becb975: [libgui.so] GUI::Application::exec() +0x25 (Application.cpp:121)
CrashDaemon(20): 0x1687adfa: [/bin/HackStudio] main +0x7ca (NonnullRefPtr.h:295)
CrashDaemon(20): 0x1687b365: [/bin/HackStudio] _start +0x45 (crt0_shared.cpp:60)
```

**Comment by minhbq-99** [comment]
This is actually the same issue as #5062 and #4700, find in files first opens files which can lead to assertion failed in Utf8View if they are binary files.

**Comment by IdanHo** [comment]
Fixed by e01f1c949fbbc7eb310e35b50f41890310af665d
